source/gateway.go

```diff
@@ -309,6 +311,11 @@ func (c *gatewayRouteResolver) resolve(rt gatewayRoute) (map[string]endpoint.Tar
 			log.Debugf("Gateway %s/%s not found for %s %s/%s", namespace, ref.Name, c.src.rtKind, meta.Namespace, meta.Name)
 			continue
 		}
+		// Confirm the Gateway has the correct name, if specified.
+		if c.src.gwName != "" && gw.gateway.Name != c.src.gwName {
```

Comment:
```suggestion
		if c.src.gwName != "" && c.src.gwName != gw.gateway.Name {
```

order of comparison (c.src.gwName first) improves readability a bit